components/movementsensor/gpsrtkserial/gps_rtk_serial.go

```diff
@@ -0,0 +1,592 @@
+// Package gpsrtkserial implements a gps using serial connection
+package gpsrtkserial
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"sync"
+
+	"github.com/de-bkg/gognss/pkg/ntrip"
+	"github.com/edaniels/golog"
+	"github.com/go-gnss/rtcm/rtcm3"
+	"github.com/golang/geo/r3"
+	slib "github.com/jacobsa/go-serial/serial"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/movementsensor"
+	gpsnmea "go.viam.com/rdk/components/movementsensor/gpsnmea"
+	rtk "go.viam.com/rdk/components/movementsensor/gpsrtk"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+)
+
+var rtkmodel = resource.DefaultModelFamily.WithModel("gps-nmea-rtk-serial")
+
+var (
+	errConnectionTypeValidation = fmt.Errorf("only serial is supported connection types for %s", rtkmodel.Name)
+	errInputProtocolValidation  = fmt.Errorf("only serial is supported input protocols for %s", rtkmodel.Name)
+)
+
+const (
+	serialStr = "serial"
+	ntripStr  = "ntrip"
+)
+
+// Config is used for converting NMEA MovementSensor with RTK capabilities config attributes.
+type Config struct {
+	NmeaDataSource           string `json:"nmea_data_source"`
+	SerialPath               string `json:"serial_path"`
+	SerialBaudRate           int    `json:"serial_baud_rate,omitempty"`
+	SerialCorrectionPath     string `json:"serial_correction_path,omitempty"`
```

Comment:
might be wrong but i think serial correction path is no longer needed now that this driver only does ntrip, I think the `serialcorrectionpath` and `serialcorrectionbaudrate`